### ISSUE TITLE
Can't log in

### ISSUE THREAD

**Issue by zzottel** [redmatrix/hubzilla#15]
Since I first set up my Hubzilla hub, today is the first time that I would have had to log in. It doesn't work, though.

I'm redirected back to to a distorted version of the login page (see screenshot).
![zottels hubzilla-server - mozilla firefox_039](https://cloud.githubusercontent.com/assets/1236369/8649252/b607d314-2968-11e5-924b-a5960434e26c.png)
Note that, in the navbar, there is "Abmelden", i.e. Logout, present. If I press that button, I'm returned to the normal-looking login page.

Looking a Firebug, I see that after posting login info, I'm first redirected to new_channel (which doesn't make sense because I already have a channel) and from there back to login.

After that, trying to fetch view/theme/redbasic/php/style.pcss is in turn redirected to login, then one successful ping, and all subsequent pings are redirected to login, too.

Trying to find out what happens here, I added some logging and found out that local_channel() never ever returns true in the numerous times it is called during the login process.

Still trying to find out how login works at all; then I might find something myself. Already posting this in case you have an idea about what might go wrong here.

**Comment by zzottel** [comment]
In the Apache logs I see, while I'm NOT yet authenticated, a successful GET on /view/theme/redbasic/php/style.pcss, directly after the POST to login, a GET to the same address results in a 302 redirect.

Similarly, a GET on /library/jgrowl/jquery.jgrowl.map while not logged in yields a 404 while not logged in, but a 302 after the POST.

**Comment by zzottel** [comment]
After deleting the PHPSESSID cookie, I'm getting a normal-looking login page after login, and no redirects anymore. I can look at directory/, channel/zottelh gives me Permission denied. As soon as I try to access network/, I'm getting redirects on style.pcss and the distorted login screen.

**Comment by zzottel** [comment]
Sorry, I give up, don't understand how the login procedure works. :-) mod/login.php only calls login(), which is defined in  boot.php and instantiates $a, which in turn calls startup()—but that's all I could find which is actually done. There must be more stuff, and I can see from the logs hat there is, but I couldn't find where everything else that seems to happen comes from, where and when the functions are called.

**Comment by zzottel** [comment]
Just to add that, the logs more or less don't show anything:

```
2015-07-13 14:06:51:52d51n988kohofs5j2urb69am3 auth.php:77:account_verify_password: password verified for hubzil[...]
2015-07-13 14:06:51:52d51n988kohofs5j2urb69am3 auth.php:275:require: authenticate: Array
(
    [account_id] => 1
    [account_parent] => 1
    [account_default_channel] => 2
    [account_salt] => d1f9336d8c8e69651f3fb5d0cef8b651
    [account_password] => f4fee[...]
    [account_email] => hubzil[...]
    [account_external] =>
    [account_language] => de
    [account_created] => 2015-07-06 09:34:31
    [account_lastlog] => 2015-07-13 13:07:29
    [account_flags] => 0
    [account_roles] => 4096
    [account_reset] =>
    [account_expires] => 0000-00-00 00:00:00
    [account_expire_notified] => 0000-00-00 00:00:00
    [account_service_class] =>
    [account_level] => 0
    [account_password_changed] => 0000-00-00 00:00:00
)
2015-07-13 14:06:52:8lh6bsgg4oa9pd1goig0q1pvj3 index.php:222:: index.php: page not found: /library/jgrowl/jquery.jgrowl.map ADDRESS: 94.[...] QUERY: q=library/jgrowl/jquery.jgrowl.map
```

That's it.

**Comment by zzottel** [comment]
Argh. Ok, solved, I can log in again.

The problem was the account_default_channel. At my first attempt at setting up my account, I tried to import my main RedMatrix channel, which failed (posts imported, but connections not, or somtheing like that, don't remember exactly). So in order not to get problems with my main RedMatrix channel, I immediately deleted the clone again.

Not sure if I first deleted the clone and then created a new channel or vice versa—the problem was that account_default_channel pointed to the deleted channel. I'll have a look how this can be prevented.
